src/swarm/neo/connection/ConnectionBase.d

```diff
@@ -1034,6 +1048,23 @@ abstract class ConnectionBase: ISelectClient
 
     /***************************************************************************
 
+        Requests to send outgoing data buffered by the OS for the socket
+        immediately. By default outgoing data are buffered to the maximum
+        payload of a TCP frame using the Linux TCP_CORK socket option.
+
+        This method is intended to be called right after `getPayloadForSending`
+        has called the `send` delegate.
+
+    ***************************************************************************/
+
+    public void flush ( )
+    {
+        this.sender.flush();
```

Comment:
I agree this needs to be better documented, but the point is following: 

There are two possibilities:

1) You write something to socket, write succeeds, you uncork the socket and you're done. This is what this case is covering.
2) You didn't manage to send anything, but you will eventually, after the line 387. Now you have data written to your socket buffer, and you want to get it out - you need to flush it at this point. 

One could argue that, if 2. is true, you don't need to call flush at this point, just at the second, but I don't think that's possible with this API - you need somehow to pass this info to `sendRequestPayload` (probably via `out` parameter or return value, as you already suggested).